hw/ip/otbn/dv/uvm/gen-binaries.py

```diff
@@ -61,60 +61,29 @@ def run(self, cmd: List[str]) -> None:
         subprocess.run(cmd, env=env)
 
 
-def take_env_line(dst: Dict[str, str], path: str, line_number: int,
-                  line: str) -> None:
-    '''Read one line from a .env file, updating dst.'''
-    line = line.split('#', 1)[0].strip()
-    if not line:
-        return
+def read_toolchain(otbn_dir: str) -> Toolchain:
+    '''Reads environment variables to get toolchain info.'''
+    env_dict = {}  # type: Dict[str, str]
 
-    parts = line.split('=', 1)
-    if len(parts) != 2:
-        raise RuntimeError('{}:{}: No equals sign in line {!r}.'.format(
-            path, line_number, line))
+    # OTBN assembler
+    env_dict['OTBN_AS'] = os.getenv('OTBN_AS')
+    if env_dict['OTBN_AS'] is None:
+        env_dict['OTBN_AS'] = f"{otbn_dir}/util/otbn_as.py"
 
-    dst[parts[0]] = parts[1]
+    # OTBN linker
+    env_dict['OTBN_LD'] = os.getenv('OTBN_LD')
+    if env_dict['OTBN_LD'] is None:
+        env_dict['OTBN_LD'] = f"{otbn_dir}/util/otbn_ld.py"
 
+    # RV32 assembler and linker
+    env_dict['RV32_TOOL_AS'] = os.getenv('RV32_TOOL_AS')
+    env_dict['RV32_TOOL_LD'] = os.getenv('RV32_TOOL_LD')
```

Comment:
When we chatted on the phone, we were talking about putting in a sensible default if these aren't specified. That way, the script is useful on the command line in the usual case where you've got the tools in your path.

Reply:
Fixed.